internal/helm/chart.go

```diff
@@ -40,21 +40,21 @@ func OverwriteChartDefaultValues(chart *helmchart.Chart, valuesFile string) (boo
 		}
 	}
 	if valuesData == nil {
-		return false, fmt.Errorf("failed to locate override values file: %s", valuesFile)
+		return fmt.Errorf("failed to locate override values file: %s", valuesFile)
 	}
 
 	// Read override values file data
 	values, err := chartutil.ReadValues(valuesData)
 	if err != nil {
-		return false, fmt.Errorf("failed to parse override values file: %s", valuesFile)
+		return fmt.Errorf("failed to parse override values file: %s", valuesFile)
 	}
 
 	// Replace current values file in Raw field
 	for _, f := range chart.Raw {
 		if f.Name == chartutil.ValuesfileName {
 			// Do nothing if contents are equal
 			if reflect.DeepEqual(f.Data, valuesData) {
-				return false, nil
+				return nil
```

Comment:
Based on this, we could still return `(bool, error)`, and decide to not package chart but use the original as an artifact (as the operation did not result in a change).

Reply:
I was thinking just about that when implementing it that way before but then had mixed feelings about it. Will take some time to re-evaluate the solution.

Reply:
I think that in general, it is great if a modifying function indicates if it actually modified anything.